src/buildApplication/fixPrerenderedRoutes.ts

```diff
@@ -0,0 +1,208 @@
+		return null;
+	}
+
+	return config;
+}
+
+/**
+ * Retrieve the path to the prerendered route, if it exists.
+ *
+ * @param config.fallback Fallback file configuration.
+ * @param dirName Directory name to use for the route.
+ * @param outputDir Vercel build output directory.
+ * @returns The path to the prerendered route, or null if it does not exist.
+ */
+async function getRoutePath(
+	{ fallback }: VercelPrerenderConfig,
+	dirName: string,
+	outputDir: string
+): Promise<string | null> {
+	const oldRoute = normalizePath(join(dirName, fallback.fsPath));
+	const oldFile = join(outputDir, 'functions', oldRoute);
+
+	// Check the prerendered file exists.
+	if (!(await validateFile(oldFile))) {
+		cliWarn(`Could not find prerendered file for ${oldRoute}`);
+		return null;
+	}
+
+	return oldFile;
+}
+
+/**
+ * Retrieve the new destination for the prerendered file, if no file already exists.
+ *
+ * @param config.fallback Fallback file configuration.
+ * @param dirName Directory name to use for the route.
+ * @param outputDir Vercel build output directory.
+ * @returns The new destination for the prerendered file, or null if a file already exists.
+ */
+async function getRouteDest(
+	{ fallback }: VercelPrerenderConfig,
+	dirName: string,
+	outputDir: string
+): Promise<{ newFile: string; newRoute: string } | null> {
+	const newRoute = normalizePath(
+		join(
+			dirName,
+			fallback.fsPath.replace(/(?:\.rsc)?\.prerender-fallback/gi, '')
+		)
+	);
+	const newFile = join(outputDir, 'static', newRoute);
+
+	// Check if a static file already exists at the new location.
+	if (await validateFile(newFile)) {
+		cliWarn(`Prerendered file already exists for ${newRoute}`);
+		return null;
+	}
+
+	return { newFile, newRoute };
+}
+
+/**
+ * Validate a prerendered route and retrieve its config file, original file path, and new destination.
+ *
+ * @param baseDir Base directory for the prerendered routes.
+ * @param file Prerendered config file name.
+ * @param outputDir Vercel build output directory.
+ * @returns Information for a valid prerendered route, or null if the route is invalid.
+ */
+async function validateRoute(baseDir: string, file: string, outputDir: string) {
+	const dirName = relative(join(outputDir, 'functions'), baseDir);
+	const config = await getRouteConfig(baseDir, file, dirName);
+	if (!config) return null;
+
+	const oldFile = await getRoutePath(config, dirName, outputDir);
+	if (!oldFile) return null;
+
+	const dest = await getRouteDest(config, dirName, outputDir);
+	if (!dest) return null;
+
+	return { config, oldFile, newFile: dest.newFile, newRoute: dest.newRoute };
+}
+
+/**
+ * Copy a prerendered file to its new destination.
+ *
+ * @param oldFile Prerendered file path.
+ * @param newFile Destination for the prerendered file.
+ */
+async function createNewRouteFile(oldFile: string, newFile: string) {
+	await mkdir(dirname(newFile), { recursive: true });
+	await copyFile(oldFile, newFile);
+}
```

Comment:
could this be made into a shared utility? (and moved to the fs file?)
```suggestion
/**
 * Copies a file from one location to another, it also creates the destination
 * directory if it doesn't exist
 *
 * @param sourceFile Original file path.
 * @param destFile Destination for the file.
 */
async function copyFileWithDir(sourceFile: string, destFile: string) {
	await mkdir(dirname(sourceFile), { recursive: true });
	await copyFile(sourceFile, destFile);
}
```

at this point it would also be great to add some unit tests for it, but if you don't want to overload the PR you can leave them out (and you or me can add them later)